jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/paths/FileSystemPool.java

```diff
@@ -0,0 +1,114 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util.paths;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * A pool of {@code zipfs} based {@code java.nio.file.FileSystem} objects to allow reference
+ * counting of open FileSystem and proper closure when the last reference count is decremented.
+ */
+public class FileSystemPool
```

Comment:
name should indicate it is for jar files:  `JarFileSystemPool` ?

Reply:
Sure, it could also be called `ZipFsFileSystemPool` (as that's what is being used)